Improve value formatting

kravets-levko opened the pull request:
1. Format values for all visualizations in the same manner, using suffixes ("thousands", "millions", etc.).
2. Make that suffixes locale-aware: use translation key `Value Formatting Scale` with following format:
`"suffix 1": scale1, "suffix 2": scale2`. Example: `"Billions": 1000000000, "Millions": 1000000, "Thousands": 1000` (order does not matter).

All features will work only with latest version of `babbage.ui` and `bubbletree` libraries.

kravets-levko committed:
openspending/openspending#1133 Improve value formatting

kravets-levko committed:
Code style

akariv commented on app/config/translations.json:
```diff
@@ -1,179 +1 @@
-{
```
Let's make sure the file is formatted with indentation and sorted keys, so that changes are tracked more easily.

akariv commented on app/front/scripts/config/config.js:
```diff
@@ -21,18 +22,26 @@ angular.module('Application')
+      // "Suffix": scale, "Suffix 2": scale2
+      var scale = $rootScope._t('Value Formatting Scale');
+      if (scale == 'Value Formatting Scale') {
+        scale = '"Billions":1000000000,"Millions":1000000,"Thousands":1000';
```
Let's move the defaults to the `visualizations.formatValue`, in case there's a parsing problem.

akariv commented on app/front/scripts/services/visualizations/index.js:
```diff
@@ -223,6 +223,57 @@ function paramsToBabbageStateRadar(params) {
+    try {
+      scale = JSON.parse('{' + scale + '}');
+    } catch (e) {
+      scale = {};
```
If we fail to parse, let's print something to the console.
Also, let's have the defaults set here and not above.

kravets-levko committed:
Merge branch 'master' of https://github.com/openspending/os-viewer into feature/better-value-formatting

kravets-levko committed:
CR

kravets-levko commented:
@akariv Code updated. I also fixed `download_translations.py` - I see that somebody added `de` translation directly into `translations.json`, so I added some code to keep such translation strings, and only update `en` and `es` from Google Spreadsheets

kravets-levko committed:
CR (small fix)
